core/templates/container_converters.h

```diff
@@ -0,0 +1,103 @@
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef CONTAINER_CONVERTERS_H
+#define CONTAINER_CONVERTERS_H
+
+#include "core/variant/typed_array.h"
+
+namespace converters {
+
+template <typename T>
+TypedArray<T> vector_to_typed_array(const Vector<T> &p_vector) {
+	TypedArray<T> ret;
+	ret.resize(p_vector.size());
+
+	Array::Iterator itr = ret.begin();
+	for (const T &E : p_vector) {
+		*itr = E;
+		++itr;
+	}
+	return ret;
+}
+
+template <typename T>
+List<T> vector_to_list(const Vector<T> &p_vector) {
+	List<T> ret;
+
+	for (const T &E : p_vector) {
+		ret.push_back(E);
+	}
+	return ret;
+}
+
+template <typename R, typename T>
+Vector<R> vector_change_type(const Vector<T> &p_vector) {
+	Vector<R> ret;
+	ret.resize(p_vector.size());
+
+	R *write = ret.ptrw();
+	for (const T &E : p_vector) {
+		*write = E;
+		write++;
+	}
+	return ret;
+}
+
+template <typename T>
+Vector<T> list_to_vector(const List<T> &p_list) {
+	Vector<T> ret;
+	ret.resize(p_list.size());
+
+	T *write = ret.ptrw();
+	for (const T &E : p_list) {
+		*write = E;
+		write++;
+	}
+	return ret;
+}
+
+template <typename R, typename T>
+TypedArray<R> list_to_typed_array_change_type(const List<T> &p_list) {
```

Comment:
How about consolidating the `_change_type` functions by introducing a `static_cast<ReturnType>` into all functions? Like so:

```c++
template <typename R, typename T>
 TypedArray<R> list_to_typed_array(const List<T> &p_list) {
}

template <typename T>
 TypedArray<T> list_to_typed_array(const List<T> &p_list) {
    return list_to_typed_array<T, T>(p_list);
}
```

called like:

```c++
Vector<int> result = list_to_typed_array(vector);
Vector<float> result = list_to_typed_array<float>(vector)
```

Reply:
That's 2 methods with the same signature, but different return type, no? Is it even allowed in C++?

Reply:
I'd have to test again, but I'm pretty sure i got something similar to work recently in another project.

Reply:
Seems like it compiles.
Although `vector_change_type()` becomes `vector_to_vector()` 🙃